tests/unit/s2n_pq_asm_noop_test.c

```diff
@@ -13,14 +13,14 @@
  * permissions and limitations under the License.
  */
 
-int main(int argc, char **argv)
+int main()
 {
     /* In CMakeLists.txt, we try_compile the PQ ASM code to determine if the
      * toolchain is compatible with the assembly instructions. Older versions
      * of CMake require that we supply a main() function in the sources that
      * we are passing to try_compile. So, in the try_compile, we use this main()
      * function as a noop. IMPORTANT NOTE: This file is referenced by name
-     * in CMakeLists.txt (which is unusual for a unit tst). If this file is
+     * in CMakeLists.txt (which is unusual for a unit test). If this file is
      * renamed, then CMakeLists.txt must be updated as well.*/
```

Comment:
This method is a little questionable, but it seems to be the best of bad options.

We need a main function, but can't really use s2nc/s2nd or any other existing tests, because then we'd have to include all dependencies in the try compile.

The other try_compiles that use inline code aren't actually `try_compile()` - they're calls to `check_c_source_compiles()`. There's no way (as far as I can tell) to mix inline C code and an assembly file within a call to `check_c_source_compiles()`. I *could* use `check_c_source_compiles()` with inline assembly (and I spent more time than I'll admit figuring this out):

```
    check_c_source_compiles("
        #include <stdint.h>
        int main() {
            uint64_t x = 3, y = 5, low, high;
            __asm__ volatile(
                \"mulx %[x], %[low], %[high]\"
                : [high] \"=r\" (high), [low] \"=r\" (low)
                : [x] \"r\" (x), \"d\" (y));
            return 0;
        }" BMI2_SUPPORTED)

    check_c_source_compiles("
        #include <stdint.h>
        int main() {
            uint32_t x = 7, y = 11;
            __asm__ volatile(
                \"adcx %[y], %[x]\"
                : [x] \"=r\" (x)
                : \"r\" (x), [y] \"rm\" (y));
            return 0;
        }" ADX_SUPPORTED)
```
but that seems way more precarious than simply relying on an empty main function in a test file.